### DISCUSSION TITLE
How can I create an share menu shortcut in KDE for this?

### DISCUSSION THREAD

**Discussion by Blue-Beaker** [localsend/localsend#1317] [Discussion (English, International)]
I'd like to create an share menu shourcut in KDE plasma for localsend to send files directly in the dolphin file explorer, plasma integration in browsers, etc. like what KDE Connect does. However I can't find command parameters for localsend to send certain file on startup. How can I make localsend send selected file at startup and add it to KDE share menu?

**Comment by Tienisto** [comment]
This is currently not implemented but it's a great idea

**Comment by DodoLeDev** [comment]
### Necroposting with a solution:
Drop this file in `~/.local/share/kio/servicemenus/` (user) or `/usr/share/kio/servicemenus/` (system).
Create the directory if it does not exist.

`localsend.desktop`
```ini
[Desktop Entry]
Type=Service
# all files inherit from application/octet-stream
MimeType=application/octet-stream;inode/directory;
Actions=sendwithlocalsend;

[Desktop Action sendwithlocalsend]
Name=Send with LocalSend
Name[fr]=Envoyer avec LocalSend
Icon=org.localsend.localsend_app

# If using local package
Exec=localsend %U

# If using Flatpak
# Exec=/usr/bin/flatpak run --branch=stable --arch=x86_64 --command=localsend --file-forwarding org.localsend.localsend_app @@u %U @@
```

Finally, give this file executable permissions.

The only drawback is that it does not appear in Share menu, but only in the Actions submenu of Dolphin

<img width="631" height="150" alt="image" src="https://github.com/user-attachments/assets/f64522e8-39e0-43ef-b3b6-d1f9c244de4f" />
